### PR TITLE
docs, test, fix : updated circular queue using array

### PR DESCRIPTION
**Pull request by Bucke200**
The current code is more efficient and more structured than the previous one, both in terms of readability and performance. I removed some unnecessary checks and added more documentation

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: operations_on_datastructures/circular_queue_using_array.cpp [hunk 1]
```diff
@@ -1,57 +1,97 @@
...
+    cout << "\nQueue elements: ";
+    // Loop through the valid elements using modular arithmetic
     for (int i = 0; i < count; i++) {
-        cout << queue[(i + front) % 10] << "\t";
+        cout << queue[(front + i) % MAX_SIZE] << "\t";
     }
 }
 
+// Main function with a menu-driven approach for queue operations
 int main() {
```
Comment (realstealthninja) [inline operations_on_datastructures/circular_queue_using_array.cpp:53]: main function should not have any code except for a call to tests

File: operations_on_datastructures/circular_queue_using_array.cpp [hunk 2]
```diff
@@ -1,57 +1,97 @@
...
+        return;
+    }
+
+    cout << "\nQueue elements: ";
+    // Loop through the valid elements using modular arithmetic
     for (int i = 0; i < count; i++) {
-        cout << queue[(i + front) % 10] << "\t";
+        cout << queue[(front + i) % MAX_SIZE] << "\t";
     }
 }
```
Comment (realstealthninja) [inline operations_on_datastructures/circular_queue_using_array.cpp:51]: add tests

File: operations_on_datastructures/circular_queue_using_array.cpp [hunk 3]
```diff
@@ -1,57 +1,97 @@
...
-void Enque(int x) {
-    if (count == 10) {
-        cout << "\nOverflow";
+// Define the maximum size of the queue.
+const int MAX_SIZE = 10;
+
+int queue[MAX_SIZE]; // Queue array
+int front = 0;       // Points to the front of the queue
+int rear = 0;        // Points to the next available position for inserting elements
+int count = 0;       // Keeps track of the number of elements in the queue
```
Comment (realstealthninja) [inline operations_on_datastructures/circular_queue_using_array.cpp:10]: queue should be a class not a set of functions

File: operations_on_datastructures/circular_queue_using_array.cpp [hunk 4]
```diff
@@ -1,57 +1,97 @@
 #include <iostream>
 using namespace std;
```
Comment (realstealthninja) [inline operations_on_datastructures/circular_queue_using_array.cpp:2]: using namespace std is bad practice and hence should be removed from the repository

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Comment by realstealthninja** [comment]
Please check the checkboxes in your pr.

**Comment by Bucke200** [comment]
ok working on it

**Comment by Bucke200** [comment]
@realstealthninja please check again and let me know if anything needs to be addressed

File: operations_on_datastructures/circular_queue_using_array.cpp [hunk 5]
```diff
@@ -1,57 +1,195 @@
...
+/**
+ * @brief A class to represent a circular queue.
+ * 
+ * This Queue class uses a fixed-size array to implement a circular queue.
+ * It supports basic operations like enqueue, dequeue, and display.
+ */
+class Queue {
+private:
+    static const int MAX_SIZE = 10; ///< The maximum number of elements the queue can hold
+    int queue[MAX_SIZE]; ///< Array to store the queue elements
```
Comment (realstealthninja) [inline operations_on_datastructures/circular_queue_using_array.cpp:14]: should use std::array

File: operations_on_datastructures/circular_queue_using_array.cpp [hunk 6]
```diff
@@ -1,57 +1,195 @@
...
-int count = 0;
+/**
+ * @brief A class to represent a circular queue.
+ * 
+ * This Queue class uses a fixed-size array to implement a circular queue.
+ * It supports basic operations like enqueue, dequeue, and display.
+ */
+class Queue {
+private:
+    static const int MAX_SIZE = 10; ///< The maximum number of elements the queue can hold
```
Comment (realstealthninja) [inline operations_on_datastructures/circular_queue_using_array.cpp:13]:
```suggestion
    static const size_t MAX_SIZE = 10; ///< The maximum number of elements the queue can hold
```

File: operations_on_datastructures/circular_queue_using_array.cpp [hunk 7]
```diff
@@ -1,57 +1,195 @@
...
-void Enque(int x) {
-    if (count == 10) {
-        cout << "\nOverflow";
+public:
+    /**
+     * @brief Constructs a new Queue object.
+     * 
+     * Initializes front, rear, and count to 0.
+     */
+    Queue();
```
Comment (realstealthninja) [inline operations_on_datastructures/circular_queue_using_array.cpp:25]: implementation and declaration should not be separated. please combine them

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: operations_on_datastructures/circular_queue_using_array.cpp [hunk 8]
```diff
@@ -1,57 +1,195 @@
...
-int front = 0;
-int rear = 0;
-int count = 0;
+/**
+ * @brief A class to represent a circular queue.
+ * 
+ * This Queue class uses a fixed-size array to implement a circular queue.
+ * It supports basic operations like enqueue, dequeue, and display.
+ */
+class Queue {
```
Comment (realstealthninja) [inline operations_on_datastructures/circular_queue_using_array.cpp:11]: this class could be generic and store all types of elements

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Pull request closed without merging**